### ISSUE TITLE
TAXII Connector Not Generating Data in Platform

### ISSUE THREAD

**Issue by The-Stuke** [OpenCTI-Platform/connectors#1288]
## Description
When querying a TAXII server the TAXII2 connector seems to be able to read the data and sends the objects to OpenCTI, but the connector never seems to generate any works and create observables. The workers attempt to read the data, but appear to error when reading the TAXII data. Giving the connector full bypass all capabilities does not solve the problem.

## Environment

1. OS: Docker
2. OpenCTI version: 5.9.0, but was not working on prior versions in the past months. Not sure when it started to fail.
3. OpenCTI client: NA

## Reproducible Steps

1. Run TAXII2 connector
2. Have it query TAXII server for new objects
3. Observe errors in logs/see no new observable created

## Expected Output

In Progress and Complete Works to be generated for the connector along with observables/indicators created.

## Actual Output

Described below

## Additional information

I have given the connector both the Connector role and the bypass all capabilities to see if it was a permission issue. Bypass all capabilities did not work.

### Logs shows objects are sent to the server
It does not matter if it is a small number of objects or large.
```
INFO Starting ping alive thread | timestamp=2023-07-10T21:07:47.956481Z name=pycti.connector
INFO Connector has never run | timestamp=2023-07-10T21:07:49.022747Z name=pycti.connector
INFO Polling entire API root Public API Root | timestamp=2023-07-10T21:07:49.997688Z name=pycti.connector
INFO Polling Collection Public Collection | timestamp=2023-07-10T21:07:50.602535Z name=pycti.connector
INFO Sending Bundle to server with '11143' objects | timestamp=2023-07-10T21:08:54.438809Z name=pycti.connector
INFO Run Complete. Sleeping until next run in 1 hours | timestamp=2023-07-10T21:08:56.240750Z name=pycti.connector
```
### Connector page showing messages present.
![image](https://github.com/OpenCTI-Platform/connectors/assets/48042955/298503cd-a5a6-4c91-bdc1-e5d517aba5e9)

### Connector info not showing any works
![image](https://github.com/OpenCTI-Platform/connectors/assets/48042955/d26bdcc7-7fa4-434b-abb9-3123de09cdb4)

### Worker Error
```
INFO Importing a indicator (id: indicator--a6b742b7-bc3d-4af9-981f-e1589c779c9f) | timestamp=2023-07-10T21:13:29.762684Z name=pycti.api
INFO Creating Indicator {viper Control Server at 139.155.2.105:60000}. | timestamp=2023-07-10T21:13:29.762915Z name=pycti.entities
ERROR Missing reference to handle creation | timestamp=2023-07-10T21:13:29.780854Z name=pycti.api
ERROR Traceback (most recent call last):
  File "/opt/opencti-worker/worker.py", line 224, in data_handler
    self.api.stix2.import_bundle_from_json(
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 209, in import_bundle_from_json
    return self.import_bundle(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 2357, in import_bundle
    self.import_object(item, update, types)
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 814, in import_object
    stix_object_results = do_import(
                          ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_indicator.py", line 531, in import_from_stix2
    return self.create(
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_indicator.py", line 385, in create
    result = self.opencti.query(
             ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/api/opencti_api_client.py", line 353, in query
    raise ValueError(
ValueError: {'name': 'MissingReferenceError', 'message': 'Missing reference to handle creation'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/opencti-worker/worker.py", line 224, in data_handler
    self.api.stix2.import_bundle_from_json(
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 209, in import_bundle_from_json
    return self.import_bundle(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 2357, in import_bundle
    self.import_object(item, update, types)
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 814, in import_object
    stix_object_results = do_import(
                          ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_indicator.py", line 531, in import_from_stix2
    return self.create(
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_indicator.py", line 385, in create
    result = self.opencti.query(
             ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/api/opencti_api_client.py", line 353, in query
    raise ValueError(
ValueError: {'name': 'MissingReferenceError', 'message': 'Missing reference to handle creation'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/opencti-worker/worker.py", line 224, in data_handler
    self.api.stix2.import_bundle_from_json(
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 209, in import_bundle_from_json
    return self.import_bundle(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 2357, in import_bundle
    self.import_object(item, update, types)
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 814, in import_object
    stix_object_results = do_import(
                          ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_indicator.py", line 531, in import_from_stix2
    return self.create(
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_indicator.py", line 385, in create
    result = self.opencti.query(
             ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/api/opencti_api_client.py", line 353, in query
    raise ValueError(
ValueError: {'name': 'MissingReferenceError', 'message': 'Missing reference to handle creation'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/opencti-worker/worker.py", line 224, in data_handler
    self.api.stix2.import_bundle_from_json(
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 209, in import_bundle_from_json
    return self.import_bundle(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 2357, in import_bundle
    self.import_object(item, update, types)
  File "/usr/local/lib/python3.11/site-packages/pycti/utils/opencti_stix2.py", line 814, in import_object
    stix_object_results = do_import(
                          ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_indicator.py", line 531, in import_from_stix2
    return self.create(
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/entities/opencti_indicator.py", line 385, in create
    result = self.opencti.query(
             ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pycti/api/opencti_api_client.py", line 353, in query
    raise ValueError(
ValueError: {'name': 'MissingReferenceError', 'message': 'Missing reference to handle creation'}
 | timestamp=2023-07-10T21:13:29.787244Z name=pycti.api
 ```

**Comment by The-Stuke** [comment]
Since I am seeing the below error for 'Missing reference to handle creation' and looking at a sample of an indicator not containing a reference in OpenCTI already. It might be a similar issue to https://github.com/OpenCTI-Platform/connectors/issues/389.

### Error message
```
ValueError: {'name': 'MissingReferenceError', 'message': 'Missing reference to handle creation'}
```

### Example indicator redacted
```json
{
      "type": "indicator",
      "spec_version": "2.1",
      "id": "indicator--81af5e1c-04b2-4865-a53a-e67b17737dda",
      "created_by_ref": "identity--bc4bc868-77f8-4692-a96e-97f95d7526c8",
      "created": "2023-07-12T08:10:02.424194Z",
      "modified": "2023-07-12T08:10:02.424194Z",
      "name": "metasploitselfsignedca Control Server at X.X.X.X",
      "description": "Malware Control Server",
      "indicator_types": [
        "command-and-control"
      ],
      "pattern": "[ipv4-addr:value = 'X.X.X.X']",
      "pattern_type": "stix",
      "pattern_version": "2.1",
      "valid_from": "2023-07-12T08:10:02.424194Z",
      "object_marking_refs": [
        "marking-definition--34098fce-860f-48ae-8e50-ebd3cc5e41da",
        "marking-definition--77bf6aa4-2787-414c-955d-6c789b482685"
      ]
    }
```

**Comment by SamuelHassine** [comment]
Fixed in 5.10. We will now let the data coming in.
